### PR TITLE
Support for docker --build-arg

### PR DESCRIPTION
**Pull request by fiolkaf**
It will be so cool if orca supports docker build-args.

We use it to pass NPM credentials, so we can fetch from private npm registry on docker build.
But it can be used for anything - HTTP_PROXY, private RubyGems creds, etc.

Sky is the limit ;-)

### CODE REVIEW

**Comment by sebastianseilund** [comment]
Sounds like a good idea :)

Could you rename `env` to `buildArgs` all the way through? I think that's more appropriate as that's where the arguments go. Other than that everything is 👍

**Comment by fiolkaf** [comment]
@sebastianseilund: are you good with it ?

File: src/watcher.js [hunk 1]
```diff
@@ -58,7 +59,12 @@ export default class Watcher {
             let tag = this.app.tag || getConfig().tag || 'orca.' + Date.now() //We append a timestamp since we want to make sure Marathon pulls the new image
             let imageName = getConfig().registry + `/${this.app.name}:${tag}`
             this.log(`Building image ${imageName}`)
-            await hostExec('docker', ['build', '-t', imageName, '.'], {cwd: this.app.dir})
+            let params = ['build', '-t', imageName, '.']
+            _.map(this.buildArgs, (value, key) => {
+                params.push('--build-arg')
+                params.push(`${key}="${value}"`)
```
Comment (sebastianseilund) [inline src/watcher.js]: Just realized that there's a risk of including bad unescaped arguments here. Can you escape both the key and the value? There must be a module for that you can use.

**Review by sebastianseilund — CHANGES_REQUESTED** [review]

File: src/watcher.js (hunk 1 above)
Comment (fiolkaf) [reply src/watcher.js]: good point, already implemented.

**Comment by sebastianseilund** [comment]
Thanks!

**Pull request merged**